amd/src/mod_hsuforum_loader.js

```diff
@@ -61,20 +61,31 @@ define(['jquery'], function ($) {
                 });
                 postObserver.observe(this, {subtree: true, childList: true});
             });
-            removeAttagging();
         }
 
     };
 
-    //ratings on chanfge hide unread
+    //ratings on change hide unread
     $('select.postratingmenu.ratinginput').change(function (){
         $(this).closest('.hsuforum-post-wrapper').find('span.hsuforum-unreadcount').hide();
     });
 
-    //remove link from @all tags
-    removeAttagging = function() {
-        $("#page-mod-hsuforum-discuss a:contains('@all')").contents().unwrap();
-    };
+    //hacky but on @all redirect to new, provision for old posts or update url
+    var replyid = $('article').attr("data-postid");
+    $.urlParam = function(name){
+        var results = new RegExp('[\?&]' + name + '=([^&#]*)').exec(window.location.href);
+        return results[1] || 0;
+    }
+    var postid = $.urlParam('d');
+
+    $(".posting a").each(function(){
+        $(this).click(function(){
```

Comment:
No need for the .each, can just bind the click on the general selector and it will bind for all of them